AutoGear.lua

```diff
@@ -242,11 +242,23 @@ if TOC_VERSION_CURRENT < TOC_VERSION_MOP then
 		if (not numTalentTabs) or (numTalentTabs < 2) then
 			AutoGearPrint("AutoGear: numTalentTabs in AutoGearGetSpec() is "..tostring(numTalentTabs),0)
 		end
-		for i = 1, numTalentTabs do
-			local spec, _, pointsSpent = GetTalentTabInfo(i)
-			if (highestPointsSpent == nil or pointsSpent > highestPointsSpent) then
-				highestPointsSpent = pointsSpent
-				highestSpec = spec
+		local _, spec, _, _, pointsSpent = GetTalentTabInfo(1)
+		if pointsSpent and pointsSpent >= 0 then
+			highestPointsSpent = pointsSpent
+			for i = 2, numTalentTabs do
+				local _, spec, _, _, pointsSpent = GetTalentTabInfo(i)
+				if (pointsSpent > highestPointsSpent) then
+					highestPointsSpent = pointsSpent
+					highestSpec = spec
+				end
+			end
+		else
```

Comment:
Reading this again, I realized there's a bug in my suggested code which you faithfully copied.  After the first `highestPointsSpent = pointsSpent`, it doesn't run `if pointsSpent > 0 then highestSpec = spec end`.  I think that line needs to be added, just to make sure that if there's at least 1 point spent in talent tree 1, that tree is marked as the spec tree.

Reply:
added highestSpec = spec

Reply:
It needs a condition of being above 0 or else it will assign highestSpec to the first talent tree even if there are 0 points in it.

Reply:
Updated

Reply:
This latest change isn't right either, because the check there can't be combined.  Checking for a number including 0 is to check for whether the value is a number at all and enters the correct block for the number of GetTalentTabInfo return values.